shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -434,7 +437,90 @@ private int getPointerDeviceTypeForToolType(int toolType) {
         }
     }
 
-    private void addPointerForIndex(MotionEvent event, int pointerIndex, int pointerChange,
+    private List<PointerAction> expandActions(MotionEvent event, int pointerIndex, int pointerChange,
```

Comment:
If you factor out `expandActions` into its own class (just like it was in the framework in converter.dart) - could you write simple unit tests then where you pass in face MotionEvents and assert that you get the expected List of PointerActions out?